### PR TITLE
Opening local dapp

### LINKED ISSUE #4031: Simplify opening Parity for local dapp
From gitter: https://gitter.im/ethcore/parity?at=586cd787d99b1dd03d2ed688

```
cd /dapps/mew
parity dapp .
```

### PR DESCRIPTION
**Pull request by tomusdrw**
Closes #4031

### CODE REVIEW

File: dapps/src/apps/fs.rs [hunk 1]
```diff
@@ -97,6 +99,19 @@ fn read_manifest(name: &str, mut path: PathBuf) -> EndpointInfo {
 		})
 }
 
+pub fn local_endpoint(path: String, signer_address: Option<(String, u16)>) -> Option<(String, Box<LocalPageEndpoint>)> {
```
Comment (rphmeier) [inline dapps/src/apps/fs.rs]: should accept `path: PathBuf`

Comment (NikVolf) [inline dapps/src/apps/fs.rs]: maybe even `local_endpoint<P: AsRef<Path>>(path: P, ...) { ... }`

Comment (tomusdrw) [inline dapps/src/apps/fs.rs]: :+1:

File: dapps/src/apps/fs.rs [hunk 2]
```diff
@@ -97,9 +99,22 @@ fn read_manifest(name: &str, mut path: PathBuf) -> EndpointInfo {
 		})
 }
 
-pub fn local_endpoints(dapps_path: String, signer_address: Option<(String, u16)>) -> Endpoints {
+pub fn local_endpoint<P: AsRef<Path>>(path: P, signer_address: Option<(String, u16)>) -> Option<(String, Box<LocalPageEndpoint>)> {
```
Comment (gavofyork) [inline dapps/src/apps/fs.rs]: would be nice to have short doc comments for non-trivial functions, even if they're non-visible outside their crates.

File: dapps/src/apps/mod.rs [hunk 3]
```diff
@@ -43,14 +44,22 @@ pub fn utils() -> Box<Endpoint> {
...
 	remote: Remote,
 	fetch: F,
 ) -> Endpoints {
 	// fetch fs dapps at first to avoid overwriting builtins
 	let mut pages = fs::local_endpoints(dapps_path, signer_address.clone());
+	for path in extra_dapps {
+		if let Some((id, endpoint)) = fs::local_endpoint(path.clone(), signer_address.clone()) {
+			pages.insert(id, endpoint);
+		} else {
+			warn!(target: "dapps", "Ingoring invalid dapp at {}", path.display());
```
Comment (gavofyork) [inline dapps/src/apps/mod.rs]: *Ignoring

**Comment by gavofyork** [comment]
typo and suggestion.

**Pull request merged**